tests/e2e/env/setup.sh

```diff
@@ -4,62 +4,66 @@ set -e
 
 . ./tests/e2e/env/shared.sh
 
-BLOG_ID=${E2E_BLOG_ID-111}
-
 if [[ -f "$E2E_ROOT/config/local.env" ]]; then
 	echo "Loading local env variables"
 	. "$E2E_ROOT/config/local.env"
 fi
 
-if [[ $FORCE_E2E_DEPS_SETUP || ! -d "$SERVER_PATH" ]]; then
-	step "Fetching server (branch ${WCP_SERVER_BRANCH-trunk})"
+BLOG_ID=${E2E_BLOG_ID-111}
 
-	if [[ -z $WCP_SERVER_REPO ]]; then
-		echo "WCP_SERVER_REPO env variable is not defined"
-		exit 1;
+# Setup WCPay local server instance.
+# Only if E2E_USE_LOCAL_SERVER is present & equals to true.
+if [[ -z $CI && "$E2E_USE_LOCAL_SERVER" != false ]]; then
+	if [[ $FORCE_E2E_DEPS_SETUP || ! -d "$SERVER_PATH" ]]; then
+		step "Fetching server (branch ${WCP_SERVER_BRANCH-trunk})"
+
+		if [[ -z $WCP_SERVER_REPO ]]; then
+			echo "WCP_SERVER_REPO env variable is not defined"
+			exit 1;
+		fi
+
+		rm -rf "$SERVER_PATH"
+		git clone --depth=1 --branch ${WCP_SERVER_BRANCH-trunk} $WCP_SERVER_REPO "$SERVER_PATH"
+	else
+		echo "Using cached server at ${SERVER_PATH}"
 	fi
 
-	rm -rf "$SERVER_PATH"
-	git clone --depth=1 --branch ${WCP_SERVER_BRANCH-trunk} $WCP_SERVER_REPO "$SERVER_PATH"
-else
-	echo "Using cached server at ${SERVER_PATH}"
-fi
+	cd "$SERVER_PATH"
+
+	step "Creating server secrets"
+	SECRETS="<?php
+	define( 'WCPAY_STRIPE_TEST_PUBLIC_KEY', '$E2E_WCPAY_STRIPE_TEST_PUBLIC_KEY' );
+	define( 'WCPAY_STRIPE_TEST_SECRET_KEY', '$E2E_WCPAY_STRIPE_TEST_SECRET_KEY' );
+	define( 'WCPAY_STRIPE_TEST_CLIENT_ID', '$E2E_WCPAY_STRIPE_TEST_CLIENT_ID' );
+	define( 'WCPAY_STRIPE_TEST_WEBHOOK_SIGNATURE_KEY', '$E2E_WCPAY_STRIPE_TEST_WEBHOOK_SIGNATURE_KEY' );
+	define( 'WCPAY_STRIPE_LIVE_PUBLIC_KEY', 'pk_live_XXXXXXX' );
+	define( 'WCPAY_STRIPE_LIVE_SECRET_KEY', 'sk_live_XXXXXXX' );
+	define( 'WCPAY_STRIPE_LIVE_CLIENT_ID', 'ca_live_XXXXXXX' );
+	define( 'WCPAY_OAUTH_ENCRYPT_KEY', str_repeat( 'a', SODIUM_CRYPTO_SECRETBOX_KEYBYTES ) );
```

Comment:
This is not related to the issue at hand, so feel free to ignore, but it's nice to fix it sooner or later:

- The test and live `CLIENT_ID`s were removed from the server, they're no longer needed.
- The `WCPAY_OAUTH_ENCRYPT_KEY` got renamed to `WCPAY_ONBOARDING_ENCRYPT_KEY`.

Reply:
Addressed in [8111110](https://github.com/Automattic/woocommerce-payments/pull/3383/commits/81111101f2f48c954f471829ba01ad5c38a29168)